File: 82/frontend/src/services/patientService.js
// src/services/patientService.js
import api from "./api";

// Get all patients
export async function getAllPatients() {
    try {
        const response = await api.get('/patients');
        const result = response.data;
        
        if (result.code === "0" && result.data) {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to get patients');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to get patients');
        }
    }
}

// Get patient by ID
export async function getPatientById(patientId) {
    try {
        const response = await api.get(`/patients/${patientId}`);
        const result = response.data;
        
        if (result.code === "0" && result.data) {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to get patient');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to get patient');
        }
    }
}

// Create a new patient
export async function createPatient(patientData) {
    try {
        const response = await api.post('/patients', patientData);
        const result = response.data;
        
        if (result.code === "0" && result.data) {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to create patient');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to create patient');
        }
    }
}

// Update patient
export async function updatePatient(patientId, patientData) {
    try {
        const response = await api.put(`/patients/${patientId}`, patientData);
        const result = response.data;
        
        if (result.code === "0" && result.data) {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to update patient');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to update patient');
        }
    }
}

// Delete patient
export async function deletePatient(patientId) {
    try {
        const response = await api.delete(`/patients/${patientId}`);
        const result = response.data;
        
        if (result.code === "0") {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to delete patient');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to delete patient');
        }
    }
}

// Get patients by organization
export async function getPatientsByOrganization(organizationId) {
    try {
        const response = await api.get(`/patients/organization/${organizationId}`);
        const result = response.data;
        
        if (result.code === "0" && result.data) {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to get patients by organization');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to get patients by organization');
        }
    }
}

// Get patients by family member/POA
export async function getPatientsByFamilyMember(familyMemberId) {
    try {
        const response = await api.get(`/patients/family-member/${familyMemberId}`);
        const result = response.data;
        
        if (result.code === "0" && result.data) {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to get patients by family member');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to get patients by family member');
        }
    }
}

// Get patient by Client ID
export async function getPatientByClientId(clientId) {
    try {
        const response = await api.get(`/patients/client/${clientId}`);
        const result = response.data;
        
        if (result.code === "0" && result.data) {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to get patient by Client ID');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to get patient by Client ID');
        }
    }
}

// Remove organization from patient
export async function removeOrganizationFromPatient(patientId, removalData) {
    try {
        const response = await api.post(`/patients/${patientId}/remove-organization`, removalData);
        const result = response.data;
        
        if (result.code === "0") {
            return {
                data: result.data
            };
        } else {
            throw new Error(result.msg || 'Failed to remove organization from patient');
        }
    } catch (error) {
        if (error.response?.data?.msg) {
            throw new Error(error.response.data.msg);
        } else if (error.message) {
            throw error;
        } else {
            throw new Error('Failed to remove organization from patient');
        }
    }
}

File: 82/backend/src/main/java/com/careapp/service/PatientService.java
package com.careapp.service;

import com.careapp.domain.Patient;
import com.careapp.repository.PatientRepository;
import org.springframework.stereotype.Service;

import javax.annotation.Resource;
import java.util.List;
import java.util.Optional;

@Service
public class PatientService {
    
    @Resource
    private PatientRepository patientRepository;
    
    // Create a new patient
    public Patient createPatient(Patient patient) {
        return patientRepository.save(patient);
    }
    
    // Get patient by ID
    public Optional<Patient> getPatientById(String id) {
        return patientRepository.findById(id);
    }
    
    // Get all patients
    public List<Patient> getAllPatients() {
        return patientRepository.findAll();
    }
    
    // Update patient
    public Patient updatePatient(Patient patient) {
        return patientRepository.save(patient);
    }
    
    // Delete patient
    public boolean deletePatient(String id) {
        if (patientRepository.existsById(id)) {
            patientRepository.deleteById(id);
            return true;
        }
        return false;
    }
    
    // Get patients by family member ID
    public List<Patient> getPatientsByFamilyMember(String familyMemberId) {
        return patientRepository.findByFamilyMemberId(familyMemberId);
    }
    
    // Get patients by POA ID
    public List<Patient> getPatientsByPOA(String poaId) {
        return patientRepository.findByPoaId(poaId);
    }
    
    // Get patient by medical record number
    public Optional<Patient> getPatientByMedicalRecordNumber(String medicalRecordNumber) {
        return patientRepository.findByMedicalRecordNumber(medicalRecordNumber);
    }
    
    // Get patient by Client ID
    public Optional<Patient> getPatientByClientId(String clientId) {
        return patientRepository.findByClientId(clientId);
    }
    
    // Assign patient to family member
    public Patient assignFamilyMember(String patientId, String familyMemberId) {
        Optional<Patient> patientOpt = patientRepository.findById(patientId);
        if (patientOpt.isPresent()) {
            Patient patient = patientOpt.get();
            patient.setFamilyMemberId(familyMemberId);
            return patientRepository.save(patient);
        }
        return null;
    }
    
    // Assign patient to POA
    public Patient assignPOA(String patientId, String poaId) {
        Optional<Patient> patientOpt = patientRepository.findById(patientId);
        if (patientOpt.isPresent()) {
            Patient patient = patientOpt.get();
            patient.setPoaId(poaId);
            return patientRepository.save(patient);
        }
        return null;
    }
    
    // Get patients by organization ID
    public List<Patient> getPatientsByOrganization(String organizationId) {
        return patientRepository.findByOrganizationId(organizationId);
    }
}